tests/test_models/test_mutables/test_derived_mutable.py

```diff
@@ -0,0 +1,188 @@
+        mc2 = OneShotMutableChannel(
+            num_channels=4, candidate_choices=[1, 4], candidate_mode='number')
+        ms = [mc1, mc2]
+
+        mc_derived = DerivedMutable.derive_concat_mutable(ms)
+        assert mc_derived.source_mutables == set(ms)
+
+        mc1.current_choice = 1
+        mc2.current_choice = 4
+        assert mc_derived.current_choice == 5
+        assert torch.equal(
+            mc_derived.current_mask,
+            torch.tensor([1, 0, 0, 1, 1, 1, 1], dtype=torch.bool))
+
+        mc1.current_choice = 1
+        mc2.current_choice = 1
+        assert mc_derived.current_choice == 2
+        assert torch.equal(
+            mc_derived.current_mask,
+            torch.tensor([1, 0, 0, 1, 0, 0, 0], dtype=torch.bool))
+
+        mv = OneShotMutableValue(value_list=[1, 2, 3])
+        ms = [mc1, mv]
+        with pytest.raises(RuntimeError):
+            _ = DerivedMutable.derive_concat_mutable(ms)
+
+    def test_mutable_channel_derived(self) -> None:
+        mc = OneShotMutableChannel(
+            num_channels=3,
+            candidate_choices=[1, 2, 3],
+            candidate_mode='number')
+        mc_derived = mc * 3
+        assert mc_derived.source_mutables == {mc}
+
+        mc.current_choice = 1
+        assert mc_derived.current_choice == 3
+        assert torch.equal(
+            mc_derived.current_mask,
+            torch.tensor([1, 1, 1, 0, 0, 0, 0, 0, 0], dtype=torch.bool))
+
+        mc.current_choice = 2
+        assert mc_derived.current_choice == 6
+        assert torch.equal(
+            mc_derived.current_mask,
+            torch.tensor([1, 1, 1, 1, 1, 1, 0, 0, 0], dtype=torch.bool))
+
+    def test_mutable_divide(self) -> None:
+        mc = OneShotMutableChannel(
+            num_channels=128,
+            candidate_choices=[112, 120, 128],
+            candidate_mode='number')
+        mc_derived = mc // 8
+        assert mc_derived.source_mutables == {mc}
+
+        mc.current_choice = 128
+        assert mc_derived.current_choice == 16
+        assert torch.equal(mc_derived.current_mask,
+                           torch.ones(16, dtype=torch.bool))
+        mc.current_choice = 120
+        assert mc_derived.current_choice == 16
+        assert torch.equal(mc_derived.current_mask,
+                           torch.ones(16, dtype=torch.bool))
+
+        mv = OneShotMutableValue(value_list=[112, 120, 128])
+        mv_derived = mv // 8
+        assert mv_derived.source_mutables == {mv}
+
+        mv.current_choice == 128
+        assert mv_derived.current_choice == 16
+        mv.current_choice == 120
+        assert mv_derived.current_choice == 16
+
+    def test_double_fixed(self) -> None:
+        choice_fn = lambda x: x  # noqa: E731
+        derived_mutable = DerivedMutable(choice_fn)
+        derived_mutable.fix_chosen({})
+
+        with pytest.raises(RuntimeError):
+            derived_mutable.fix_chosen({})
+
+    def test_source_mutables(self) -> None:
```

Comment:
Add a UT when the link path is longer than 1. For example, 
`A` is a source mutable, `B` is a source mutable, `C` is derived from `A`, `D` is derived from `C` and `B`. Will `D` changes along with `A` and `B`? How about it's source mutables?

Reply:
Add [`test_nested_mutables`](https://github.com/open-mmlab/mmrazor/pull/215/files#diff-c697a0e22a3d96de082519250e753c5376f3ff7d96558822fbcb5c9b4dad4aebR167)